defaults/main.yml

```diff
@@ -1,7 +1,56 @@
 ---
 
+## Role variables ##
+
+ka_resolvconf_install: yes
+
+## Kontena main Variables ##
+
+# Kontena Master websocket uri (required)
 ka_master_uri: "ws://127.0.0.1:8080"
+
+# Kontena Grid token (required)
 ka_token: "secret_token"
+
+# network interface for peer/private communication (default: eth1)
 ka_peer_interface: "eth1"
 
-ka_config_path: "{{ (ansible_distribution_release == 'xenial') | ternary('/etc/kontena-agent.env', '/etc/default/kontena-agent') }}"
+
+
+## Kontena other variables ##
+
+# Specify node public ip, overrides default resolving
```

Comment:
I did not see default strategy for public ip resolving . Did you miss calling http://docs.ansible.com/ansible/latest/ipify_facts_module.html maybe?

Reply:
Automatically ip determination for that... hmm, so, it interesting. Okay.

Reply:
I propose to add flag, for example "ka_public_ip_auto_resolve". If it is set. ansible will resolve it in automatically way and  assign it as ka_public_ip parameter. It will allow to provision nodes without manual configuration for public ip.

Reply:
Yes, of course. I also thought.